includes/compat/trait-wc-stripe-subscriptions.php

```diff
@@ -674,19 +682,40 @@ private function create_mandate_options_for_order( $order, $subscriptions ) {
 			return [];
 		}
 
-		// If this is the first order, not a renewal, then get the subscriptions for the parent order.
-		if ( empty( $subscriptions ) ) {
-			$subscriptions = wcs_get_subscriptions_for_order( $order );
-		}
+		$sub_amount = 0;
 
-		// If there are no subscriptions we just return since mandates aren't required.
-		if ( 0 === count( $subscriptions ) ) {
-			return [];
-		}
+		// If this is a switch order we set the mandate options based on the new subscription.
+		$cart_contain_switches = WC_Subscriptions_Switcher::cart_contains_switches();
+		if ( $cart_contain_switches ) {
+			foreach ( WC()->cart->cart_contents as $cart_item ) {
+				$sub_amount += (int) WC_Subscriptions_Product::get_price( $cart_item['data'] );
```

Comment:
We may need to consider different decimal currencies like [this](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/compat/trait-wc-stripe-subscriptions.php#L688-L690)?

Reply:
Good catch, Mayisha! I fixed it here and also [here](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3359/files#diff-516e329017ff60072191226cd6dfc80b4b9ca8ccccacd6c151c1598cf4179434R712) (which ignored the currency).